DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc

```diff
@@ -1124,6 +1136,40 @@ void SiPixelTrackResidualSource::analyze(const edm::Event& iEvent, const edm::Ev
 		float x = clustgp.x(); 
 		float y = clustgp.y(); 
 		float z = clustgp.z();
+
+		float xclust = clust->x();
+                float yclust = clust->y();
+
+                int pxfpanel     = tTopo->pxfPanel((*hit).geographicalId());
+                int pxfmodule    = tTopo->pxfModule((*hit).geographicalId());
+                int pxfdisk      = tTopo->pxfDisk((*hit).geographicalId());
+                int pxfblade_off = tTopo->pxfBlade((*hit).geographicalId());
+
+                // translate to online conventions
+
+		if (z<0.) { pxfdisk  = -1.*pxfdisk; }
+                int pxfblade = -99;
+                if (pxfblade_off<=6 && pxfblade_off>=1)        { pxfblade = 7-pxfblade_off;  }
+                else if (pxfblade_off<=18 && pxfblade_off>=7)  { pxfblade = 6-pxfblade_off;  }
+                else if (pxfblade_off<=24 && pxfblade_off>=19) { pxfblade = 31-pxfblade_off; }
+
+                int clu_sdpx = ((pxfdisk>0) ? 1 : -1) * (2 * (abs(pxfdisk) - 1) + pxfpanel);
+                int binselx = (pxfpanel==1&&(pxfmodule==1||pxfmodule==4)) ? (pxfmodule==1) : ((pxfpanel==1&& xclust<80.0)||(pxfpanel==2&&xclust>=80.0));
+                int nperpan = 2 * pxfmodule + pxfpanel - 1 + binselx;
+                int clu_roc_binx = ((pxfdisk>0) ? nperpan : 9 - nperpan) + (clu_sdpx + 4) * 8 - 2 * ((abs(pxfdisk)==1) ? pxfdisk : 0);
+
+                int clu_roc_biny = -99.;
+                int nrocly = pxfmodule + pxfpanel;
+                for (int i=0; i<nrocly; i++) {
+                  int j = (pxfdisk<0) ? i : nrocly - 1 - i;
+                  if (yclust>=(j*52.0)&& yclust<((j+1)*52.0))
+                    clu_roc_biny = 6 - nrocly + 2 * i + ((pxfblade>0) ? pxfblade-1 : pxfblade + 12)*12 + 1;
+                }
+                if (pxfblade>0) { clu_roc_biny = clu_roc_biny+144; }
+
+                meRocBladevsDiskEndcap->setBinContent(clu_roc_binx,clu_roc_biny, meRocBladevsDiskEndcap->getBinContent(clu_roc_binx,clu_roc_biny)+1);
```

Comment:
@gouskos if you use the correct function from here http://cmslxr.fnal.gov/source/DQMServices/Core/interface/MonitorElement.h#0337 you can get the root object (TH2) and use the AddBinContent function on it. Hope this helps.